sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -122,9 +122,10 @@ class queue_impl {
 
   void throw_asynchronous() {
     if (m_AsyncHandler && m_Exceptions.size()) {
-      m_AsyncHandler(m_Exceptions);
+      exception_list Exceptions;
+      std::swap(m_Exceptions, Exceptions);
+      m_AsyncHandler(Exceptions);
```

Comment:
Why not
m_AsyncHandler(std::move(Exceptions));
?

Reply:
Done, thanks!

Reply:
What about just
m_AsyncHandler(std::move(**m_**Exceptions));
?

Reply:
Disregard my latest comment.